packages/core/src/AbstractRenderer.ts

```diff
@@ -37,17 +83,17 @@ export class AbstractRenderer extends EventEmitter
      * @param {number} [options.backgroundColor=0x000000] - The background color of the rendered area
      *  (shown if not transparent).
      */
-    constructor(system, options)
+    constructor(type: RENDERER_TYPE = RENDERER_TYPE.UNKNOWN, options?: IRendererOptions)
     {
         super();
 
         // Add the default render options
         options = Object.assign({}, settings.RENDER_OPTIONS, options);
 
         // Deprecation notice for renderer roundPixels option
-        if (options.roundPixels)
+        if ((options as IRendererOptionsLegacy).roundPixels)
         {
-            settings.ROUND_PIXELS = options.roundPixels;
+            settings.ROUND_PIXELS = (options as IRendererOptionsLegacy).roundPixels;
```

Comment:
I am not very fond of this double casting but I also don't really see how to deal with this better.

Reply:
It is deprecated property, it is not existed on `IRendererOptions` by default and used only for back support.
We can't use it on public api. Casting is normal usage for this case.